File: 프로그래머스/3/43162. 네트워크/네트워크.java
import java.util.*;

class Solution {
    public void check(int target, int n, int[][] computers, boolean[] computer) {
        computer[target] = false;
        
        for (int i = 0; i < n; i++) {
            if (i != target && computers[target][i] == 1 && computer[i]) {
                check(i, n, computers, computer);
            }
        }
    }
    
    public int solution(int n, int[][] computers) {
        int answer = 0;
        
        boolean[] computer = new boolean[n];
        Arrays.fill(computer, true);
        
        for(int i = 0; i < n; i++) {
           if (computer[i]) {
               answer += 1;
               check(i, n, computers, computer);
           }
        }
        return answer;
    }
}

File: 프로그래머스/3/43162. 네트워크/네트워크.py
def check(target, n, computers, computer):
    computer[target] = False
    
    for i in range(n):
        if i != target and computers[target][i] == 1 and computer[i]:
            check(i, n, computers, computer)
            

def solution(n, computers):
    computer = [True] * n
    answer = 0
    
    for i in range(n):
        if computer[i]:
            answer += 1
            check(i, n, computers, computer)
            
    return answer